### PR TITLE
feat: Add a generic `Application.UnhandledException` handler

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Currently some issues go unnoticed quietly under the rader

## What is the new behavior?

In debug, the issues are always logged and debugger breaks in case `DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION` is not defined (in-line with WinUI)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
While in line with WinUI, I am not sure whether `UnhandledException` is not occurring too often, so much that it could disrupt users. In such cases we have options:

- Remove the `Debugger.Break` for now completely
- Define the `DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION` constant by default for Uno targets

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 1]
```diff
@@ -563,6 +565,24 @@ void ApplyLiteralProperties()
 
 				writer.AppendLineIndented(";");
 			}
+
+			void AttachUnhandledExceptionHandler()
+			{
+				writer.AppendLineIndented($"#if DEBUG");
+				writer.AppendLineIndented($"UnhandledException += (s, e) =>");
+				writer.AppendLineIndented("{");
+				writer.Indent();
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:574]: You could do this as `writer.BlockInvariant("UnhandledException += (s, e) =>")`

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 2]
```diff
@@ -563,6 +565,24 @@ void ApplyLiteralProperties()
 
 				writer.AppendLineIndented(";");
 			}
+
+			void AttachUnhandledExceptionHandler()
+			{
+				writer.AppendLineIndented($"#if DEBUG");
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:571]: Usually the formatting of `#if`s always start at the beginning of the line. So I think it should be unindented

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 3]
```diff
@@ -563,6 +565,24 @@ void ApplyLiteralProperties()
...
+				using (writer.BlockInvariant("if (global::System.Diagnostics.Debugger.IsAttached)"))
+				{
+					writer.AppendLineIndented("global::System.Diagnostics.Debug.WriteLine(e.Exception);");
+					writer.AppendLineIndented("#if !DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION");
+					writer.AppendLineIndented("global::System.Diagnostics.Debugger.Break();");
+					writer.AppendLineIndented("#endif");
+				}
+				writer.Indent(-1);
+				writer.AppendLineIndented("};");
+				writer.AppendLineIndented($"#endif");
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:584]: Same here

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs (hunk 1 above)
Comment (Youssef1313) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:574]: Ignore this comment. `BlockInvariant` may force the semicolon to be in a separate line. We could keep the code more verbose for better formatting

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 4]
```diff
@@ -563,6 +565,24 @@ void ApplyLiteralProperties()
...
+			void AttachUnhandledExceptionHandler()
+			{
+				writer.AppendLineIndented($"#if DEBUG");
+				writer.AppendLineIndented($"UnhandledException += (s, e) =>");
+				writer.AppendLineIndented("{");
+				writer.Indent();
+				using (writer.BlockInvariant("if (global::System.Diagnostics.Debugger.IsAttached)"))
+				{
+					writer.AppendLineIndented("global::System.Diagnostics.Debug.WriteLine(e.Exception);");
+					writer.AppendLineIndented("#if !DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION");
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs]: Added docs

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**